c/tskit/tables.c

```diff
@@ -267,7 +267,7 @@ tsk_individual_table_expand_location(
         = TSK_MAX(additional_length, self->max_location_length_increment);
     tsk_size_t new_size = self->max_location_length + increment;
 
-    if (check_offset_overflow(self->location_length, increment)) {
+    if (check_offset_overflow(self->max_location_length, increment)) {
```

Comment:
Since we are checking `max_location_length` instead of `location_length` then the check should go inside of the next `if` statement, where we actually expand `max_location_length`, shouldn't it?

Reply:
Yes! You're right. As it stands now it will error out on the extension one before the overflow.

Reply:
Note that this is assuming that `location_length` is always less than `max_location_length`, so we don't need to check for overflow on the former. (which is true)

Reply:
You have to be careful though, as if `self->metadata_length` overflows you don't expand the column at all and crash.

Reply:
So do we need another check on `location_length`? This is only checking `max_location_length`.